core/include/kernel/dt.h

```diff
@@ -324,6 +324,41 @@ void reinit_manifest_dt(void);
 /* Returns TOS_FW_CONFIG DTB or SP manifest DTB if present, otherwise NULL */
 void *get_manifest_dt(void);
 
+/*
+ * Find the offset of a parent node in the parent node cache
+ * @fdt: FDT to work on
+ * @node_offset: Offset of the node we look for its parent
+ * @parent_offset: Output parent node offset upon success
+ * @return 0 on success and -1 on failure
+ *
+ * This function is supported when CFG_DT_CACHED_NODE_INFO is enabled.
```

Comment:
I believe the cache feature should be more transparent. Ideally hook the cache into the FDT lib code. Or if we don't want to touch libfdt, make a wrapper. But the caller should not need to know if the cache is enabled or not.

Reply:
I'm fine patch _libfdt_ but I'm pretty confident such change would never hit libfdt mainstream.
The way `fdt_reg_base_address()` is implemented in our _kernel/dt.h_ make that fully leverage adresse cell (to prevent a node property parsing) makes that this very function needs to be patched. 
Same for `fdt_reg_size()` (to prevent 2 parsing of node properties) and `fdt_fill_device_info()` (2 parent node lookup + 2 node properties parsing).

That said, I can change this inline description to:
```patch
- * Find the offset of a parent node in the parent node cache
+ * Find the offset of a parent node in the parent node cache if any
 * (...)
- *
- * This function is supported when CFG_DT_CACHED_NODE_INFO is enabled.

Reply:
I've removed the fixup commit that held your comment. The commit was noisy and buggy so I it replaced with another fixup commit that should work better.
Please feel free to use this comment thread if needed. I won't tag it 'Resolved' :)